### DISCUSSION TITLE
keep variants

### DISCUSSION THREAD

**Discussion by fshndev** [google/model-viewer#4088] [Q&A] [answered]
https://user-images.githubusercontent.com/113173416/216141698-4a5a30cc-7cee-4220-98a8-4f569236d58d.mp4

can you help with this problem? I tried many times but couldn't find a solution. I found the second file on github.

**Comment by elalish** [comment, marked as answer]
Thanks for the video, but what were you expecting to see vs. what you got? And how did you create your model? Can you attach your model?

**Reply by fshndev** [reply]
https://github.com/google/model-viewer/discussions/3822
Example: https://cashcredenza.s3.us-east-2.amazonaws.com/index.html
-I want to make the example, but I'm having trouble with the variant part.

-When I switch from variant 1 to 6 it works fine but when I switch from variant 6 to 1 it removes variant 6 and only variant 1 remains.

-There is no such problem in the sample project. I looked for a long time but couldn't find a solution.

**Reply by fshndev** [reply]
.

**Reply by fshndev** [reply]
You can better understand my problem if we examine the models with the model viewer.

models link: https://we.tl/t-RsJms5ZbGL (2 glb file)

**Reply by fshndev** [reply]
![image](https://user-images.githubusercontent.com/113173416/216155761-3df654a1-6ea1-486b-9be3-f00eda4ae7c6.png)
(wetransfer link content)

**Reply by danilokrulj** [reply]
Not sure if I fully understand but I think you are expecting a change in a certain model part when selecting a variant instead of change in the whole model. You expect the change from one variant to apply to another variant. Basically, I think you have to make every possibility as a variant. If you make a variant where the left block is black and the middle one and the right one are white they will be white each time you select that variant.

**Reply by fshndev** [reply]
I want to change from a specific partition. Like you, I thought of doing all the options as variants. Later I saw that it was not necessary. A separate variant was not created for each of them in this project.

- https://cashcredenza.s3.us-east-2.amazonaws.com/index.html

**Reply by elalish** [reply]
@danilokrulj has it right. glTF variants are each for the entire model - they cannot be combined. If you want to mix and match materials, you're better of skipping variants and using our [materials API](https://modelviewer.dev/examples/scenegraph/#pickMaterialExample) to change the individual material properties instead.

**Reply by fshndev** [reply]
Got it, thanks for everything.

**Reply by danilokrulj** [reply]
@elalish turns out you can make variants for each material, only not through the editor I assume , a colleague exported variants from blender and it works as the example shown at the start of the discussion. 

[variant.webm](https://user-images.githubusercontent.com/92090684/222167144-3d9ea648-9fe1-4f49-8fbf-063116b4647c.webm)

**Comment by fshndev** [comment]
https://user-images.githubusercontent.com/113173416/216344282-9435fe96-b7b7-4bef-a5c3-081d82486ff6.mp4

I hope this video was clearer.
